velox/core/PlanNode.h

```diff
@@ -1782,4 +1782,60 @@ class WindowNode : public PlanNode {
   const RowTypePtr outputType_;
 };
 
+/// The MarkDistinct operator is used to produce aggregate mask channels for
+/// distinct aggregations. The mask channel is a boolean vector set to true for
+/// input all rows that contain new unique combination of grouping keys and
+/// distinct aggregate column values.
+/// @param markerVariable Name of the output mask channel.
+/// @param distinctVariables Names of grouping keys and distinct aggregate
+/// column.
+/// @param hashVariable Optional. Name of channel that contains pre-computed
+/// hash of grouping keys and distinct aggregate column.
+class MarkDistinctNode : public PlanNode {
+ public:
+  MarkDistinctNode(
+      PlanNodeId id,
+      FieldAccessTypedExprPtr markerVariable,
```

Comment:
This is an output column, right? It should be specified by name. See AggregationNode::aggregateNames and ProjectNode::names.

Reply:
Can you double check? I think if we use AggregationNode as a baseline, similar columns in AggregationNode are FieldAccessTypedExprPtr.

```
AggregationNode(
      const PlanNodeId& id,
      Step step,
      const std::vector<FieldAccessTypedExprPtr>& groupingKeys,
      const std::vector<FieldAccessTypedExprPtr>& preGroupedKeys,
      const std::vector<std::string>& aggregateNames,
      const std::vector<CallTypedExprPtr>& aggregates,
      const std::vector<FieldAccessTypedExprPtr>& aggregateMasks,
      bool ignoreNullKeys,
      PlanNodePtr source);
```

Reply:
For example, markerVariable becomes aggregateMasks in real world query plan.

Reply:
@shiyu-bytedance Input columns are specified as FieldAccessTypedExprPtr. Output columns are specified using column name. E.g. aggregateNames specify output columns. I assume markerVariable indicates an output column and is not an input column. Hence, it should be identified by name.